truffleruby/src/main/c/cext/ruby.c

```diff
@@ -748,9 +823,14 @@ VALUE rb_big_cmp(VALUE x, VALUE y) {
 
 void rb_big_pack(VALUE val, unsigned long *buf, long num_longs) {
   long i;
-  VALUE longs = truffle_invoke(RUBY_CEXT, "rb_big_pack", val, num_longs);
+  long j;
+  VALUE bytes = truffle_invoke(RUBY_CEXT, "rb_integer_bytes", val, (int)num_longs, 8, Qfalse, Qtrue, Qfalse);
   for (i = 0; i < num_longs; i++) {
-    buf[i] = (unsigned long)truffle_read_idx_l(longs, i);
+    unsigned long x = 0;
+    for(j = 0; j < 8; j++) {
+      x = x | ((unsigned long)(truffle_read_idx_i(bytes, 8 * i + j) & 0xff)) << (8 * j);
+    }
+    buf[i] = x;
```

Comment:
Yeah. I should have revisited that. The initial work to get `rb_integer_bytes` working was done by replacing the guts of `rb_big_pack` and I only did the work on `rb_integer_pack` after that.